streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -36,7 +36,11 @@
  * Join operations requires that topics of the joining entities are copartitoned, i.e., being partitioned by the same key and having the same
  * number of partitions. Copartitioning is ensured by having the same number of partitions on
  * joined topics, and by using the serialization and Producer's default partitioner.
+ *
+ * @deprecated since 2.4 release; will be removed in 3.0.0 via KAFKA-7785
  */
+// TODO: move this class into an internal package instead of deleting it (cf KAFKA-7785)
```

Comment:
Since we have a Jira ticket that is even referenced here, I would prefer to remove the ToDo from the code.